### PR TITLE
Alternative stack/pool allocator implementation based on Cray pointers compatible with Cray+AMD stack

### PR DESCRIPTION
**Pull request by MichaelSt98**
… pointers but which works on Cray + AMD

Some problems with OMNI related to the unit tests ...

### CODE REVIEW

File: transformations/transformations/pool_allocator.py [hunk 1]
```diff
@@ -93,6 +175,10 @@ class TemporariesPoolAllocatorTransformation(Transformation):
     check_bounds : bool, optional
         Insert bounds-checks in the kernel to make sure the allocated stack size is not
         exceeded (default: `True`)
+    cray_ptr_loc_rhs : bool, optional
+        Whether to only pass the stack variable as integer to the kernel(s) or 
+        whether to pass the whole stack array to the driver and the calls to ``LOC()`` 
+        within the kernel(s) itself (default: `False`)
```
Comment (reuterbal) [inline transformations/transformations/pool_allocator.py:181]:
```suggestion
        When enabled, pass the full stack array from the driver to the kernel and perform
        the calls to ``LOC()`` within the kernel(s) to determine the "pointer address"
        of the temporary arrays. By default, ``LOC()`` will only be called in the
        driver and "pointer addresses" are passed as arguments to the kernel
        (default: ``False``).
```

File: transformations/transformations/pool_allocator.py [hunk 2]
```diff
@@ -725,7 +857,7 @@ def create_pool_allocator(self, routine, stack_size):
         if loop_map:
             routine.body = Transformer(loop_map).visit(routine.body)
 
-    def inject_pool_allocator_into_calls(self, routine, targets, ignore):
+    def inject_pool_allocator_into_calls(self, routine, targets, ignore, driver=False):
```
Comment (reuterbal) [inline transformations/transformations/pool_allocator.py:860]: The `driver` argument seems unnecessary here?

**Review by reuterbal — COMMENTED** [review]
Many thanks!

I have rebased the branch and cannot find any specific OMNI failures anymore. Disregard the currently failing CLOUDSC test, which will be fixed with ecmwf-ifs/dwarf-p-cloudsc#82. I have locally verified this.

The implementation looks good to me apart from two minor comments. I still need to test this on NVIDIA and AMD platforms, though, and assess the impact. Therefore, only a comment for now.

**Review by reuterbal — APPROVED** [review]
Tested on Nvidia, too. Performance seems unharmed.

**Comment by MichaelSt98** [comment]
Looks good to me. Thanks for fixing the OMNI related stuff and for testing!

**Pull request merged**